### DISCUSSION TITLE
Ava is unable to run when `NODE_OPTIONS=--use-openssl-ca` is used globally.

### DISCUSSION THREAD

**Discussion by Jym77** [avajs/ava#3148] [Q&A] [answered]
My company is having some network proxy/firewall with an OpenSSL certificate. In order to tell node to use it, for example for downloading packages, I'm having a 
```shell
export NODE_OPTIONS=--use-openssl-ca
```
in my global shell config files, which works fine.

But when I try to run Ava, I now get the following error:
```
 ✘ Internal error
  Error [ERR_WORKER_INVALID_EXEC_ARGV]: Initiated Worker with invalid NODE_OPTIONS env variable: --use-openssl-ca is not allowed in NODE_OPTIONS
  Error [ERR_WORKER_INVALID_EXEC_ARGV]: Initiated Worker with invalid NODE_OPTIONS env variable: --use-openssl-ca is not allowed in NODE_OPTIONS
      at new NodeError (node:internal/errors:371:5)
      at new Worker (node:internal/worker:194:13)
      at createWorker (file:///c/Projects/a11y/alfa-act-r/node_modules/ava/lib/fork.js:24:12)
      at loadFork (file:///c/Projects/a11y/alfa-act-r/node_modules/ava/lib/fork.js:84:39)
      at pMap.concurrency.concurrency (file:///c/Projects/a11y/alfa-act-r/node_modules/ava/lib/api.js:289:20)
      at file:///c/Projects/a11y/alfa-act-r/node_modules/p-map/index.js:141:26
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
Removing the option from `NODE_OPTIONS` let it work fine.

Given the error message and a tiny bit of digging, I believe this is similar to https://github.com/parcel-bundler/parcel/issues/6569
Sounds like a worker is forked with the global `NODE_OPTIONS` being forwarded, but in a context where `--use-openssl-ca` makes no sense.

My current workaround is to have two terminals open, one with the option set to install packages or any other task requiring network access in node, and another one with the option unset to run Ava 🙈 

---
Please share relevant sample code. Or better yet, provide a link to a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).

This happens in [This public repo](https://github.com/Siteimprove/alfa-act-r) when running `yarn test` (after `yarn install` and `yarn build`), and the `NODE_OPTIONS=--use-openssl-ca` is set.
Sorry I'm not having time to build a better MNWE right now 😕 

---
We'll also need your AVA configuration (in `package.json` or `ava.config.*` configuration files) and how you're invoking AVA. Share the installed AVA version (get it by running `npx ava --version`).

Ava version 5.1.0 (was also here with 4.3.1),
```
"ava": {
    "verbose": true,
    "files": [
      "test/**/*.spec.js"
    ]
  },
```

**Comment by novemberborn** [comment, marked as answer]
You can set `"workerThreads": "false"` in the AVA config, this should work with child processes. Perhaps setting `"environmentVariables": { "NODE_OPTIONS": "" }` could also work, to override them for the worker threads.

**Reply by Jym77** [reply]
`workerThreads` didn't work for me, but using `environmentVariables` to overwrite `NODE_OPTIONS` did the trick 👍
